src/components/modes/teacher/BackgroundToggle.js

```diff
@@ -25,26 +26,51 @@ const useStyles = makeStyles(() => ({
 const BackgroundToggle = () => {
   const classes = useStyles();
   const { t } = useTranslation();
-  const dispatch = useDispatch();
 
-  const { backgroundImage } = useSelector(
-    ({ appInstance }) => appInstance.content.settings,
+  const { mutate: postAppSetting } = useMutation(
+    MUTATION_KEYS.POST_APP_SETTING,
   );
 
-  const toggleDisabled = !backgroundImage?.uri;
+  const { mutate: patchAppSetting } = useMutation(
+    MUTATION_KEYS.PATCH_APP_SETTING,
+  );
+
+  const [ backgroundToggleSetting, setBackgroundToggleSetting ] = useState(null);
+  
+  const { data: appSettings, isSuccess } = useAppSettings();
+
+  useEffect(() => {
+    if(isSuccess) {
+      const backgroundSetting = appSettings?.find(
+        ({ name }) => name === APP_SETTINGS.BACKGROUND,
+      );
+      if(backgroundSetting){
+        setBackgroundToggleSetting(appSettings?.find(
+          ({ name }) => name === APP_SETTINGS.BACKGROUND_TOGGLE,
+        ) || {
+          name: APP_SETTINGS.BACKGROUND_TOGGLE,
+          data: {
+            toggle: false,
+          }
+        });
+      }
+    }
+  }, [appSettings, isSuccess]);
+
+  const toggleDisabled = backgroundToggleSetting === null;
```

Comment:
Wasn't this preventing the user to toggle if the setting wasn't fetch? I think you don't really need it anymore, or you should check with `isLoading` of app settings.

Reply:
Actually, in the case where the backgroundSetting is not set (no background image was uploaded), I would the toggle to be disabled. But I can add the `isLoading` check as a security.